x/gov/keeper/proposal.go

```diff
@@ -95,6 +95,54 @@ func (keeper Keeper) SubmitProposal(ctx sdk.Context, messages []sdk.Msg, metadat
 	return proposal, nil
 }
 
+// CancelProposal will cancel proposal before the voting period ends
+func (keeper Keeper) CancelProposal(ctx sdk.Context, proposalID uint64, proposer string) error {
+	proposal, ok := keeper.GetProposal(ctx, proposalID)
+	if !ok {
+		return sdkerrors.Wrapf(types.ErrProposalNotFound, "proposal not found with %d id", proposalID)
+	}
+
+	if proposal.Proposer == "" {
+		return sdkerrors.Wrapf(types.ErrInvalidProposal, "proposal %d does n't have proposer %s, so cannot be canceled", proposalID, proposer)
```

Comment:
Can we add a comment on this for explaining the reason of this check? It is obvious when following this PR discussion, but if you look at the code only you can wonder why it is needed as the proposer is set.

```suggestion
		return sdkerrors.Wrapf(types.ErrInvalidProposal, "proposal %d doesn't have proposer %s, so cannot be canceled", proposalID, proposer)
```